nameko/containers.py

```diff
@@ -127,9 +139,14 @@ def __init__(self, service_cls, worker_ctx_cls, config):
         self.max_workers = (
             config.get(MAX_WORKERS_CONFIG_KEY) or DEFAULT_MAX_WORKERS)
 
-        self.dependencies = DependencySet()
-        for dep in prepare_dependencies(self):
-            self.dependencies.add(dep)
+        # the relationship between extensions and the container is symbiotic;
+        # containers are responsible for starting and stopping their
+        # extensions, and extensions are able to interact with the container
+        # via its :attr:`ServiceContainer.interface`.
+        self.dependencies = ExtensionSet()
+        for name, ext in discover_extensions(service_cls):
+            bound = ext.bind(name, self.interface)
```

Comment:
Nope. This was as straightforward as I could make it with the container knowing about all extension instances. I can't believe I didn't think about using the superclass!